feat: technologies table

toddbaert opened the pull request:
Transforms SDK table into new "technologies" table, which showcases SDKs + install instructions + providers + hooks.

Direct link: https://deploy-preview-77--lucky-creponne-baf9b3.netlify.app/docs/reference/technologies/ (click on any of the languages).

beeme1mr commented on src/components/TechnologiesFeatures/GolangFeatures/index.tsx:
```diff
@@ -0,0 +1,34 @@
+        href: 'https://github.com/open-feature/golang-sdk'
+      }}
+      artifact={{
+        title: 'Github repository',
```
```suggestion
        title: 'Go Package',
```

beeme1mr commented on src/components/TechnologiesFeatures/GolangFeatures/index.tsx:
```diff
@@ -0,0 +1,34 @@
+      }}
+      artifact={{
+        title: 'Github repository',
+        href: 'https://github.com/open-feature/golang-sdk',
```
```suggestion
        href: 'https://pkg.go.dev/github.com/open-feature/golang-sdk',
```

benjiro commented on src/components/TechnologiesFeatures/DotnetFeatures/index.tsx:
```diff
@@ -0,0 +1,25 @@
+      svg={CSharpNoFillSvg}
+      sdkRepoLink={{
+        title: 'Github repository',
+        href: 'https://github.com/open-feature/golang-sdk'
```
```suggestion
        href: 'https://github.com/open-feature/dotnet-sdk'
```

benjiro commented on src/components/TechnologiesFeatures/DotnetFeatures/index.tsx:
```diff
@@ -0,0 +1,25 @@
+      artifact={{
+        title: 'Github repository',
+        href: 'https://github.com/open-feature/golang-sdk',
+        instruction: 'Install-Package OpenFeature',
```
I think these days most people would be using the cross-platform dotnet cli instead of the powershell methods.
```suggestion
        instruction: 'dotnet add package OpenFeature',
```

benjiro commented on src/components/TechnologiesFeatures/DotnetFeatures/index.tsx:
```diff
@@ -0,0 +1,25 @@
+      }}
+      artifact={{
+        title: 'Github repository',
+        href: 'https://github.com/open-feature/golang-sdk',
```
```suggestion
        href: 'https://www.nuget.org/packages/OpenFeature',
```

toddbaert commented:
@beeme1mr @benjiro Thanks for catching my copy/paste mistakes! :sweat_smile:

beeme1mr commented on src/components/custom/OpenFeatureTechnologiesPage/index.tsx:
```diff
@@ -0,0 +1,86 @@
+      <OpenFeatureComponentTable items={ props.providers } />
+
+      <h3>Hooks</h3>
+      <p>Available {props.technology} hooks. Hooks add behavior at well-defined points during the <i>flag evaluation lifecycle</i>.</p>
```
What do you think about something like this?

Hooks add behavior at well-defined points during the [flag evaluation lifecycle](/docs/reference/concepts/hooks/#flag-evaluation-life-cycle). The following hooks are available for {props.technology}.

beeme1mr replied:
It's basically a reordering of what you had.

toddbaert replied:
Ya I think the reordering just seems to emphasize the technology in a weird way. I'll play with it a bit more.

toddbaert replied:
I've basically done this, but made the link relative.

beeme1mr reviewed (commented):
It also looks like the some of the icons don't dim when you hover over them. The most obvious one is the "add yours" tile.

toddbaert commented on src/components/custom/OpenFeatureTechnologiesPage/index.tsx:
```diff
@@ -0,0 +1,86 @@
+      </p>
+      
+      <h3>Providers</h3>
+      <p>Available {props.technology} providers. Providers connect the <i>evaluation API</i> which <i>application authors</i> use, to a backend that resolves flag values.</p>
```
I like the first sentence but not the second, the technology is obvious at this point.

beeme1mr replied:
That's fine. The first sentence was the main part I wanted to update.

toddbaert replied:
I've basically done this, but made the link relative.

toddbaert commented on src/components/TechnologiesFeatures/JavascriptFeatures/index.tsx:
```diff
@@ -0,0 +1,41 @@
+          description: 'A provider for flagd',
+          href: 'https://github.com/open-feature/node-sdk-contrib/tree/main/libs/providers/flagd',
+          svg: FlagdSvg
+        }
```
Let's do that in another PR.

toddbaert commented:
> It also looks like the some of the icons don't dim when you hover over them. The most obvious one is the "add yours" tile.

Great catch. I extracted the "svg or fonticon" logic the cards use for use elsewhere and this broke. I've fixed it.

beeme1mr reviewed (approved):
Love it, thanks!
